### PR TITLE
Price decrease

### PR DESCRIPTION
**Pull request by DonatasPuc**
Implemented price decrease. I don't know if it is the optimal solution, but currently the client can pick price decrease type (either "AMOUNT" or "PERCENT") by which current price is calculated and returned in a FoodResponse dto.

### CODE REVIEW

File: backend/Persistence/DB/Data/Test/Foods.json [hunk 1]
```diff
@@ -1,150 +1,218 @@
 [
   {
-    "Price": 9.25,
-    "CreatedAt": "2008-06-12T00:00:00",
+    "StartingPrice": 15,
+    "CreatedAt": "2021-11-04T11:47:04.9955968+02:00",
+    "IdRestaurant": "0",
+    "TypesOfFood": [],
+    "StartDecreasingAt": "2021-11-04T11:47:04.9955968+02:00",
+    "DecreaseType": "PERCENT",
```
Comment (audrius-savickas) [inline backend/Persistence/DB/Data/Test/Foods.json:8]: I think we should use enums for `DecreaseType`, for example 0 for percent type, and 1 for value type

File: backend/Persistence/DB/Data/Test/Foods.json [hunk 2]
```diff
@@ -164,7 +232,30 @@
         "Name": "Drink"
       }
     ],
+    "StartDecreasingAt": "2021-11-04T11:47:04.9955968+02:00",
+    "DecreaseType": "PERCENT",
```
Comment (audrius-savickas) [inline backend/Persistence/DB/Data/Test/Foods.json:236]: foods.json should have foods with decrease type of value as well, not only percent like right now

File: backend/Persistence/DB/Data/Test/Foods.json [hunk 3]
```diff
@@ -164,7 +232,30 @@
...
+    "CreatedAt": "2021-11-04T09:59:21.56Z",
+    "IdRestaurant": "0",
+    "TypesOfFood": [
+      {
+        "Id": "0",
+        "Name": "Unclassified"
+      }
+    ],
+    "StartDecreasingAt": "2021-11-04T09:59:21.56Z",
+    "DecreaseType": "string",
```
Comment (audrius-savickas) [inline backend/Persistence/DB/Data/Test/Foods.json:254]: string?

File: backend/Domain/Entities/Food.cs [hunk 4]
```diff
@@ -5,28 +5,99 @@ namespace Domain.Entities
...
         public string IdRestaurant { get; set; }
         public virtual IEnumerable<TypeOfFood> TypesOfFood { get; set; }
+        public DateTime StartDecreasingAt { get; set; }
+        public string DecreaseType { get; set; }
+        public double IntervalTimeInMinutes { get; set; }
+        public decimal AmountPerInterval { get; set; }
+        public double PercentPerInterval { get; set; }
+
+        private const string AMOUNT_STR = "AMOUNT";
+        private const string PERCENT_STR = "PERCENT";
```
Comment (audrius-savickas) [inline backend/Domain/Entities/Food.cs:19]: move these to enum

**Review by audrius-savickas — CHANGES_REQUESTED** [review]
I'm not sure if this exception throwing in `Food.cs` constructor is the most effective, I think you can improve the readability a bit

**Comment by DonatasPuc** [comment]
Changed DecreaseType to enum. Added validation to api's for validating DecreaseType enum value. Improved data in joods.json.

Refactored Food constructor, but I still don't know if it's ok.

Also, where should I document what enum values mean in the web api?

**Review by DominykasSip — APPROVED** [review]

**Review by JesusGonzalezA — APPROVED** [review]

**Comment by JesusGonzalezA** [comment]
> Also, where should I document what enum values mean in the web API?

We should, but we would have to import new packages or create new classes just to alter the documentation UI. I believe it takes time and it does not really give us a huge increase of value. For me it is correct leaving it alone for now

**Pull request merged**